lib/resty/shell.lua

```diff
@@ -126,6 +126,22 @@ function _M.run(cmd, stdin, timeout, max_size)
     local reason, status
     ok, reason, status = proc:wait()
 
+    if ok == nil and reason ~= "exited" then
+        err = reason
+        local ok2, err2 = cleanup_proc(proc)
+        if not ok2 then
+            err = concat_err(err, err2)
+        end
+
+        local stdout = concat(stdout_tab)
+        release_tab(tab_pool_tag, stdout_tab)
+
+        local stderr = concat(stderr_tab)
+        release_tab(tab_pool_tag, stderr_tab)
+
+        return nil, stdout, stderr, "failed to wait process: " .. tostring(err)
```

Comment:
It would me more grammarly correct to say "failed to wait _for_ process". Should we update this error message before the release?

Reply:
We should.